src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -202,16 +191,17 @@ export default class MongooseAdapter {
     const plural = this.inflector.plural;
 
     // Set up some data structures based on resourcesOrCollection
-    const resourceTypes = [];
-    const changeSets = {};
+    let resourceTypes = [];
+    let changeSets = {};
+
     const idOrIds = mapResources(resourceOrCollection, (it) => {
       changeSets[it.id] = it;
       resourceTypes.push(it.type);
       return it.id;
     });
 
-    const mode    = typeof idOrIds === "string" ? "findOne" : "find";
-    const idQuery = typeof idOrIds === "string" ? idOrIds : {"$in": idOrIds};
+    let idQuery = this.constructor.makeIdQuery(idOrIds);
+    let mode = typeof idQuery === 'string' ? 'findOne' : 'find';
```

Comment:
To answer your questions, I really don't know. I'm guessing I just deleted the two lines, rewrote them, and didn't really think about it.

I love that suggestion though. Replaced.